AdobeStockImage/Test/Integration/Model/SaveImageTest.php

```diff
@@ -0,0 +1,213 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\AdobeStockImage\Test\Integration\Model;
+
+use Magento\AdobeStockAssetApi\Api\AssetRepositoryInterface;
+use Magento\AdobeStockImageApi\Api\SaveImageInterface;
+use Magento\Framework\Api\AttributeValueFactory;
+use Magento\Framework\Api\Search\Document;
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\Framework\App\Filesystem\DirectoryList;
+use Magento\Framework\Exception\IntegrationException;
+use Magento\Framework\Filesystem;
+use Magento\Framework\Filesystem\Driver\Https;
+use Magento\Framework\Filesystem\DriverInterface;
+use Magento\TestFramework\Helper\Bootstrap;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Test client for communication to Adobe Stock API.
+ */
+class SaveImageTest extends TestCase
+{
+    const URL_FIELD = 'thumbnail_240_url';
+
+    /**
+     * @var SaveImageInterface
+     */
+    private $saveImage;
+
+    /**
+     * @var DriverInterface
+     */
+    private $driver;
+
+    /**
+     * @var Filesystem
+     */
+    private $fileSystem;
+
+    /**
+     * @var AssetRepositoryInterface
+     */
+    private $assetRepository;
+
+    /**
+     * @var SearchCriteriaBuilder
+     */
+    private $criteriaBuilder;
+
+    /**
+     * @return array
+     */
+    public function getSaveTestDataProvider(): array
+    {
+        return [
+            'image_save' => [
+                'documentData' => [
+                    'id' => 1,
+                    'comp_url' => 'https://test.url/magento-logo.png',
+                    'width' => 110,
+                    'title' => 'test',
+                    'content_type' => 'image/png',
+                    'height' => 210,
+                    'some_bool_param' => false,
+                    'some_nullable_param' => null,
+                    'extension_attributes' => [
+                        'title' => 'test',
+                        'is_downloaded' => 0,
+                        'is_licensed_locally' => 0,
+                        'thumbnail_240_url' => 'https://test.url/magento-logo.png',
+                        'creator_id' => 1122,
+                        'creator_name' => 'Test',
+                        'path' => 'catalog/category/tmp.png',
+                        'content_type' => 'image/png',
+                        'category' => [
+                            'id' => 1,
+                            'name' => 'Test'
+                        ],
+                    ]
+                ],
+                'sourcePath' => 'magento-logo.png',
+                'destinationPath' => 'catalog/category/tmp.png',
```

Comment:
Let's give a more unique name to the test image
```suggestion
                'destinationPath' => 'catalog/category/adobe-stock-save-image-test.png',
```